### PR TITLE
Support GSU for FP16/BF16 HPA source kernel

### PR DESCRIPTION
**Pull request by jichangjichang**

### CODE REVIEW

File: Tensile/KernelWriterSource.py [hunk 1]
```diff
@@ -340,6 +340,8 @@ def functionPrefix(self, kernel):
     for i in range(1, kernel["ProblemType"]["NumIndicesC"]):
       indexChar = self.indexChars[i]
       kStr += " + (IDX%s)*strideD%s" % (indexChar, indexChar)
+    if kernel["_GlobalAccumulation"] == 'MultipleBuffer':
+      kStr += " + (gsuSumIdx)*strideW"
```
Comment (TonyYHsieh) [inline Tensile/KernelWriterSource.py:344]: strideW is not defined in GLOBAL_D macro.

**Review by TonyYHsieh — APPROVED** [review]
Looks good.

File: Tensile/KernelWriterSource.py (hunk 1 above)
Comment (jichangjichang) [reply Tensile/KernelWriterSource.py:344]: same design with other strideXXX

**Review by imcarsonliao — APPROVED** [review]

**Comment by jichangjichang** [comment]
@AlexBrownAMD @zaliu 
Could you review this PR?
We would like to enable HIP GSU for NT Size [256,1,1,32768] on ROCM3.10 .
untrained kernel takes 4400us to run this size.
Trained hip gsu kernel takes 90us to run this size.

**Comment by jichangjichang** [comment]
@AlexBrownAMD  we decided to merge it first since rocm3.10 deadline is coming. If you have comments or suggestions for this PR, I can modify it by another PR.

**Review by AlexBrownAMD — COMMENTED** [review]
lgtm

**Pull request merged**